### ISSUE TITLE
Watch Dogs 2 flickering & graphical corruption on Windows/AMD Proprietary Drivers

### ISSUE THREAD

**Issue by koffiato** [doitsujin/dxvk#4146]
There's significant amount of flickering whenever any 3D object is rendered; menus work just fine. [Here's a video of the said issue.](https://youtu.be/1-Hpv5cO3CA)

### Software information
Game is the Ubisoft Connect version, using "-nosplash -eac_launcher" launch options. Also very lightly modded (removed intro videos for the sake of convenience). In game video settings and/or mods do not matter, issue happens regardless.

### System information
- GPU: AMD Radeon RX 6700 XT
- Driver: 24.6.1
- OS version: Windows 11 23H2
- DXVK version: 2.4.0

### Apitrace file(s)
- [WatchDogs2.trace](https://drive.google.com/file/d/1GFos_QwynC3LZUploroFdB01RBXp0h9R/view?usp=sharing)

### Log files
- [WatchDogs2_d3d11.log](https://github.com/user-attachments/files/16229140/WatchDogs2_d3d11.log)
- [WatchDogs2_dxgi.log](https://github.com/user-attachments/files/16229141/WatchDogs2_dxgi.log)

**Comment by Blisto91** [comment]
Hi there. 
Just for info. This is actually game bug that it is not currently known why dxvk can trigger on some hardware.

https://www.reddit.com/r/watch_dogs/comments/1d0wa4r/nvidiawatch_dogs_2_skyboxshadow_flicker_fix/
https://www.pcgamingwiki.com/wiki/Watch_Dogs_2#Nvidia_RTX_3000_and_4000_series_flickering_issues

**Comment by koffiato** [comment]
Interesting. Just tested the game on RADV, it's working just fine with it. I remember playing the game with 23.8.2 drivers as well. Will try with some random dxvk.conf's (hiding AMD and whatnot).

**Comment by HunterCZ122** [comment]
I can reproduce this issue on RADV with Mesa 24.1.3.

**Comment by HunterCZ122** [comment]
I found someone on ProtonDB with the same issue on RDNA3 card too, so I think the flickering on RADV applies to RDNA3, but not on RDNA2 and older architectures.

**Comment by rustygear29** [comment]
> I found someone on ProtonDB with the same issue on RDNA3 card too, so I think the flickering on RADV applies to RDNA3, but not on RDNA2 and older architectures.

I can't find the RDNA3 option and does he even work at all? 

-7700 
-R 3600
-ram: 32gb 3200

**Comment by HunterCZ122** [comment]
> I can't find the RDNA3 option and does he even work at all? 

RDNA3 is a GPU architecture (mostly RX 7000 series).

> -gpu: 7700 

If you meant RX 7700 XT,  then you have an RDNA3 card.
Do you have the same issue? If not, then this feels like a regression (if you use Mesa version provided by Linux Mint)

**Comment by GalecO1987** [comment]
Hello, I can reproduce this issue as well with my RX 7700 XT and Mesa 24.1.5.

https://github.com/user-attachments/assets/c1636182-da00-4b8f-8632-04dfcf86e079

**Comment by GalecO1987** [comment]
Also there are more reports about this issue:

- https://www.protondb.com/app/447040#KvXJGed2P2

- https://www.reddit.com/r/linux_gaming/comments/1dc5jdc/watch_dogs_flickering/

**Comment by Blisto91** [comment]
Yes we know. I can reproduce on my 7900xtx too, but the main d3d11 dev @doitsujin haven't so far which would be needed to look at it

**Comment by rustygear29** [comment]
> Hello, I can reproduce this issue as well with my RX 7700 XT and Mesa 24.1.5.
> 
>  Video_2024-08-08_14-43-37.1.mp4

On the options my GPU is unidentified I can't identify the RX 7700 XT but has (RADV).

**Comment by rustygear29** [comment]
> > I can't find the RDNA3 option and does he even work at all?
> 
> RDNA3 is a GPU architecture (mostly RX 7000 series).
> 
> > -gpu: 7700
> 
> If you meant RX 7700 XT, then you have an RDNA3 card. Do you have the same issue? If not, then this feels like a regression (if you use Mesa version provided by Linux Mint)

Everything's fine I just can't find the option to switch from RDNA3 to RDNA2 and I could when I started the game first I could switch or at least he gave me option to still can't find it for some dumbass reason or I'm blind or there was no option to begin with.

**Comment by rustygear29** [comment]
> Also there are more reports about this issue, both with AMD 7xxx cards:
> 
> * https://www.protondb.com/app/447040#KvXJGed2P2
> * https://www.reddit.com/r/linux_gaming/comments/1dc5jdc/watch_dogs_flickering/

https://www.reddit.com/r/linux_gaming/comments/1emxs8c/amd_linux_watch_dogs_2_skyboxflickering_issue/
(Add my post)

**Comment by FourH3ad** [comment]
https://github.com/user-attachments/assets/85c84385-65a3-41f5-a08c-47f9cbb39b78

Can confirm this issue happens on RDNA3 amd gpus.

GPU: AMD Radeon RX 7900 XTX (radeonsi, navi31, LLVM 18.1.8, DRM 3.57, 6.10.5-arch1-1)
GPU DRIVER: 4.6 Mesa 24.1.6-arch1.1

**Comment by Blisto91** [comment]
Can any of you test if the issue is gone with this dxvk build? Download button is at the bottom
https://github.com/doitsujin/dxvk/actions/runs/13554104427

**Comment by doitsujin** [comment]
merged the pr now, should hopefully be fixed. It's a game bug either way though.

**Comment by kyngs** [comment]
Hey, sorry for necroposting. Just tried to play WD2 with this patch, and I'm having issues with the game running unexpectedly slow. I'm talking 1440p 60FPS on very high on an RX 7900XTX. Is it possible that this patch caused such a performance degradation, or should I look elsewhere?

**Comment by Blisto91** [comment]
Do you mean versus dxvk without the workaround or versus native d3d11?

**Comment by HellsingMatt** [comment]
can confirm this issue still happening with dxvk 2.6.2, Win11 24H2, Ryzen5 7500F + RX7650gre

and with a huge fps drop, 60+fps --> 20-fps

**Comment by Blisto91** [comment]
Attach the logs that appear next to the exe

**Comment by HellsingMatt** [comment]
[WatchDogs2_d3d11.log](https://github.com/user-attachments/files/21058667/WatchDogs2_d3d11.log)
[WatchDogs2_dxgi.log](https://github.com/user-attachments/files/21058668/WatchDogs2_dxgi.log)

**Comment by HellsingMatt** [comment]
[WatchDogs2_dxgi.log](https://github.com/user-attachments/files/21075153/WatchDogs2_dxgi.log)
[WatchDogs2_d3d11.log](https://github.com/user-attachments/files/21075154/WatchDogs2_d3d11.log)

![Image](https://github.com/user-attachments/assets/76b2691f-8026-4866-9c59-fd167d28bba2)

https://github.com/user-attachments/assets/c15bfc20-b1a7-4576-8eb6-2916e02086ae

**Comment by Squall-Leonhart** [comment]
@HellsingMatt are you still experiencing this regression on Proprietary driver 2.0.353 (25.8.1+)

**Comment by HellsingMatt** [comment]
> [@HellsingMatt](https://github.com/HellsingMatt) are you still experiencing this regression on Proprietary driver 2.0.353 (25.8.1+)

@Squall-Leonhart  i've figured out this issue is caused by fog, can be fixed by disable fog effect in game's graphic options
